ad_source/view_sets.py

```diff
@@ -195,7 +195,9 @@ class UserTasks(views.APIView):
 
     def get(self, request):
         tasks = Task.objects.filter(user=request.user)
-        serializer = serializers.TaskSerializer(instance=tasks, context={'request': request}, many=True)
+        contract_address = request.query_params.get('contract_address')
```

Comment:
If the parameter has to be mandatory, I would suggest to change `urls.py` to:
```py
path(r'api/user/tasks/<str:contract_address>/', ...)
```

and function descriptor to:
```py
def get(self, request, *, contract_address: str):
```

..as I've noticed unit-tests where missing (shame on me), hope you don't mind if I'll add them myself and change the url config :ok_hand: